agenta-backend/agenta_backend/services/llm_apps_service.py

```diff
@@ -1,37 +1,56 @@
 import asyncio
+import json
 import logging
-from typing import Any, List
-
-from agenta_backend.models.api.evaluation_model import AppOutput
+from typing import Any, Dict, List
 
 import httpx
 
+from agenta_backend.models.api.evaluation_model import AppOutput
 
 # Set logger
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
 
-async def get_llm_app_output(uri: str, datapoint: Any, parameters: dict) -> AppOutput:
-    prompt_user = replace_placeholders(parameters["prompt_user"], datapoint)
-    prompt_system = replace_placeholders(parameters["prompt_system"], datapoint)
+async def invoke_app(
+    uri: str, datapoint: Any, parameters: Dict, openapi_parameters: List[Dict]
+) -> AppOutput:
+    """
+    Invokes an app for one datapoint.
+    Uses the openapi_parameters from the openapi.json to determine how to invoke the app
+
+    Args:
+        uri (str): The URI of the app to invoke.
+        datapoint (Any): The data to be sent to the app.
+        parameters (Dict): The parameters required by the app taken from the db.
+        openapi_parameters (List[Dict]): The OpenAPI parameters of the app.
+
+    Returns:
+        AppOutput: The output of the app.
+
+    Raises:
+        httpx.HTTPError: If the POST request fails.
+    """
 
     url = f"{uri}/generate"
 
-    payload = {
-        "temperature": parameters["temperature"],
-        "model": parameters["model"],
-        "max_tokens": parameters["max_tokens"],
-        "prompt_system": prompt_system,
-        "prompt_user": prompt_user,
-        "top_p": parameters["top_p"],
-        "frequence_penalty": parameters["frequence_penalty"],
-        "presence_penalty": parameters["presence_penalty"],
-        "inputs": {
-            input_item["name"]: datapoint.get(input_item["name"], "")
-            for input_item in parameters["inputs"]
-        },
-    }
+    payload = {}
+    inputs_dict = {}
```

Comment:
Can we have this in a separated method, it tends to be more complex in the future depending on the param_type.